### PR TITLE
Track like actions

### PR DESCRIPTION
**Pull request by basine**
... and some refactoring

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift [hunk 1]
```diff
@@ -38,16 +38,32 @@ public extension ConversationCell {
...
-    @objc public func likeMessage(button: AnyObject!) {
+    @objc public func likeMessage(sender: AnyObject!) {
         guard message.canBeLiked else { return }
         self.messageToolboxView.forceShowTimestamp = false
+        let reactionType : ReactionType = message.liked ? .Unlike : .Like
+        trackReaction(sender, reaction: reactionType)
 
-        ZMUserSession.sharedSession().performChanges {
-            self.message.liked = !self.message.liked
+        ZMUserSession.sharedSession().performChanges { [weak self] in
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift]: Curious, why it is necessary to weakify self here?

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2079,6 +2082,9 @@
 		F5EAB12A1B120665008DB66A /* SketchBottomView.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; name = SketchBottomView.m; path = Sketchpad/SketchBottomView.m; sourceTree = "<group>"; };
 		F5F52F701B11D7050053F1C8 /* SketchViewController.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; name = SketchViewController.h; path = Sketchpad/SketchViewController.h; sourceTree = "<group>"; };
 		F5F52F711B11D7050053F1C8 /* SketchViewController.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; name = SketchViewController.m; path = Sketchpad/SketchViewController.m; sourceTree = "<group>"; };
+		F93D58421D7D9184003AE972 /* Analytics+ConversationEvents+Swift.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = "Analytics+ConversationEvents+Swift.swift"; sourceTree = "<group>"; };
```
Comment (mikeger) [inline Wire-iOS.xcodeproj/project.pbxproj]: I think that the file extension is already implying that the file is a swift addition.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift (hunk 1 above)
Comment (marcoconti83) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift]: It's not necessary, but we talked about this in general and said that it does not hurt to put weakify in blocks as a habit, even if it's not needed in this point specifically.

File: Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+extension ZMMessage {
+    
+    public class func analyticsAttribute(message: ZMConversationMessage) -> String {
+        let messageType = Message.messageType(message)
+        return analyticsAttribute(withMessageType: messageType)
+    }
+    
+    public class func analyticsAttribute(withMessageType messageType: MessageType) -> String {
```
Comment (mikeger) [inline Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift]: For me `analyticsAttribute` sounds too broad, I would prefer to have a more specific name, like `analyticsType`.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift (hunk 1 above)
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift]: Talked -- but not with me, I see no point in this.

File: Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        case .Unknown:      return "unknown"
+        case .Text:         return "text"
+        case .Image:        return "image"
+        case .Audio:        return "audio"
+        case .Video:        return "video"
+        case .RichMedia:    return "rich_media"
+        case .Ping:         return "ping"
+        case .File:         return "file"
+        case .System:       return "system"
+        case .Location:     return "location"
```
Comment (mikeger) [inline Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift]: Would be a bit more elegant to have a mapping dict MessageType->String, but already a good improvement.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift (hunk 1 above)
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift]: I thought the outcome was the we should make it clear in the documentation of `performChanges` that it's fine not use weak.

File: Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift (hunk 3 above)
Comment (basine) [inline Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift]: Yeah, but we return a String, so `type` is also weird.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift (hunk 1 above)
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Like.swift]: The comment was added for `NSManagedObjectContext.performGroupedBlock(_:)`, but not yet for the equivalent user session methods.

File: Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift (hunk 3 above)
Comment (mikeger) [inline Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift]: You are right, then may be `analyticsTypeString` ?

**Comment by daehn** [comment]
💣 conflicts

File: Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift (hunk 4 above)
Comment (basine) [inline Wire-iOS/Sources/Analytics/Events/ZMMessage+Analytics.swift]: Well in the optimal case we would only use Swift and use a String enum for `MessageType`. I think that would make sense to use `MessageType` only internally in Analytics, call the tracking methods with the message instead and do the mapping in a private method.

**Pull request merged**